trade-finance-manager-api/src/v1/controllers/utilisation-reports/fee-record-correction/put-fee-record-correction-transient-form-data.controller.test.ts

```diff
@@ -0,0 +1,97 @@
+import httpMocks from 'node-mocks-http';
+import { AxiosResponse, HttpStatusCode, AxiosError } from 'axios';
+import api from '../../../api';
+import { aTfmSessionUser } from '../../../../../test-helpers';
+import {
+  putFeeRecordCorrectionTransientFormData,
+  PutFeeRecordCorrectionTransientFormDataRequest,
+} from './put-fee-record-correction-transient-form-data.controller';
+
+console.error = jest.fn();
+
+jest.mock('../../../api');
+
+describe('put-fee-record-correction-transient-form-data.controller', () => {
+  describe('putFeeRecordCorrectionTransientFormData', () => {
+    afterEach(() => {
+      jest.resetAllMocks();
+    });
+
+    const reportId = '1';
+    const feeRecordId = '2';
+    const formData = {};
+    const user = aTfmSessionUser();
+
+    const getHttpMocks = () =>
+      httpMocks.createMocks<PutFeeRecordCorrectionTransientFormDataRequest>({
+        params: { reportId, feeRecordId },
+        body: {
+          formData,
+          user,
+        },
+      });
+
+    it(`should return a ${HttpStatusCode.Ok} status code if the api request is successful`, async () => {
+      // Arrange
+      const { req, res } = getHttpMocks();
+
+      // Act
+      await putFeeRecordCorrectionTransientFormData(req, res);
+
+      // Assert
+      expect(res._getStatusCode()).toEqual(HttpStatusCode.Ok);
+      expect(res._isEndCalled()).toEqual(true);
+
+      expect(api.updateFeeRecordCorrectionTransientFormData).toHaveBeenCalledTimes(1);
```

Comment:
could these be split into different its too please

Reply:
Split out thanks